src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java

```diff
@@ -185,88 +186,101 @@ private List<Element<BundlingRequestDocument>> mapSystemGeneratedcaseDocument(Li
+                fileNameBuilder.append("_" + DateFormatHelper.formatLocalDate(
+                    witnessDocs.getValue()
+                        .getWitnessOptionUploadDate(),
+                    "ddMMyyyy"
+                ));
+            }
+            Document document = witnessDocs.getValue().getWitnessOptionDocument();
+            bundlingWitnessDocs.add(BundlingRequestDocument.builder()
+                                        .documentFileName(fileNameBuilder.toString())
+                                        .documentLink(DocumentLink.builder()
+                                                          .documentUrl(document.getDocumentUrl())
+                                                          .documentBinaryUrl(document.getDocumentBinaryUrl())
+                                                          .documentFilename(document.getDocumentFileName()).build())
+                                        .build());
+
+        });
         return ElementUtils.wrapElements(bundlingWitnessDocs);
     }
 
     private List<Element<BundlingRequestDocument>> mapUploadEvidenceExpertDoc(List<Element<UploadEvidenceExpert>> uploadEvidenceExpert, String displayName) {
         List<BundlingRequestDocument> bundlingExpertDocs = new ArrayList<>();
-        if (!Optional.ofNullable(uploadEvidenceExpert).isEmpty()) {
-            uploadEvidenceExpert.forEach(expertDocs -> {
-                StringBuilder fileNameBuilder = new StringBuilder();
-                fileNameBuilder.append(displayName);
-                if (Optional.ofNullable(expertDocs.getValue().getExpertOptionName()).isPresent()) {
-                    fileNameBuilder.append("_" + expertDocs.getValue().getExpertOptionName());
-                }
-                if (Optional.ofNullable(expertDocs.getValue().getExpertOptionUploadDate()).isPresent()) {
-                    fileNameBuilder.append("_" + DateFormatHelper.formatLocalDate(expertDocs.getValue()
-                                                                                      .getExpertOptionUploadDate(), "ddMMyyyy"));
-                }
-                Document document = expertDocs.getValue().getExpertDocument();
-                bundlingExpertDocs.add(BundlingRequestDocument.builder()
-                                           .documentFileName(fileNameBuilder.toString())
-                                           .documentLink(DocumentLink.builder()
-                                                             .documentUrl(document.getDocumentUrl())
-                                                             .documentBinaryUrl(document.getDocumentBinaryUrl())
-                                                             .documentFilename(document.getDocumentFileName()).build())
-                                           .build());
-
-            });
+        if (null == uploadEvidenceExpert || uploadEvidenceExpert.isEmpty()) {
```

Comment:
You don't need the `isEmpty()` check on uploadEvidenceExpert. If the collection is empty, the forEach just won't do anything.

Can be changed as follows:
```suggestion
        if (null == uploadEvidenceExpert) {
```